### DISCUSSION TITLE
How to force webpack to use "node:" prefixes?

### DISCUSSION THREAD

**Discussion by krutoo** [webpack/webpack#18265] [Q&A] [answered]
Hi, im trying to build node app and i need to tell webpack to use `node:` specifiers in its internal code.

For example in result file i see:
```
import { createRequire as __WEBPACK_EXTERNAL_createRequire } from "module";
```

I want to change such specifiers from `"module"` to `"node:module"`.

How can i do this?

**Comment by alexander-akait** [comment]
Hi, we don't have options for this, why do you need it? Both work fine with Node.js

**Reply by krutoo** [reply]
@alexander-akait I am trying to bundle application for run in Deno =)

(I apologize for the incorrect description)

**Reply by alexander-akait** [reply, marked as answer]
Got it, we need to implement `nodePrefixForCoreModules` here https://github.com/webpack/webpack/blob/main/lib/config/browserslistTargetHandler.js#L336, put version there, than implement `supportsModePrefixForCoreModules() {}` in `RuntimeTemplate.js` and use it everywhere we generate external exports for core modules, do you want to send a PR, if yes - feel free to do it, if no - feel free to open an issue, I will try to implement it tomorrow (easy task)

**Reply by alexander-akait** [reply]
So we will use them when Node.js support it (if you set for example `target: "node18"`) or you set `output.environment.nodePrefixForCoreModules: true`

**Reply by krutoo** [reply]
@alexander-akait Thank you a lot for answer, I opened an issue: https://github.com/webpack/webpack/issues/18277
